File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_map = dict()
        t_map = dict()

        for i in range(len(s)):
            s_char = s[i]
            t_char = t[i]
            if s_char not in s_map:
                s_map[s_char] = 0
            if t_char not in t_map:
                t_map[t_char] = 0
            
            s_map[s_char] += 1
            t_map[t_char] += 1

        return s_map == t_map

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        
        int[] diff = new int[26];
        
        for (int i = 0; i < s.length(); ++i) {
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);
            
            ++diff[sChar - 'a'];
            --diff[tChar - 'a'];
        }
        
        for (int i = 0; i < diff.length; ++i) {
            if (diff[i] != 0) {
                return false;
            }
        }
        
        return true;
    }
}